.github/workflows/php.yml

```diff
@@ -0,0 +1,33 @@
+name: PHP Composer
+
+on:
+  push:
+    branches: [ "**" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v3
```

Comment:
You can use cache v4, like we've done for the provision libraries, eg -> https://github.com/upmind-automation/provision-provider-domain-names/blob/main/.github/workflows/static-analysis.yml